README.md

```diff
@@ -1,91 +1,551 @@
+    ```
+
+    This installs the pgvector and pgai extensions on your local PostgreSQL developer
+    environment, then installs the package dependencies in your Python environment.
+
+   
+1. Create the pgai extension in a database. Use either:
+
+   - SQL:
+      1. Connect to PostgreSQL:
+         ```bash
+         psql -d "postgres://postgres:pgaipass@localhost:9876/postgres"
+         ```
+         TODO: Is this true for the local installation? Is the database already created?
+
+      1. Create the pgai extension:
+   
+          ```sql
+          CREATE EXTENSION IF NOT EXISTS ai CASCADE;
+          ```
+    
+          The `CASCADE` automatically installs the plpython3u and pgvector dependencies.
+
+   - Terminal: 
+     1. In `Makefile`, update `DB` and `USER` to match your PostgreSQL configuration. 
+     1.  Create the pgai extension:
+
+        ```bash
+        make create_extension
+        ```
+</details>
+
+
+## Securely connect to OpenAI through pgai
+
+API keys are secrets. Exposing them can present financial and information security issues. 
+The following sections show how to securely connect to your OpenAI account through pgai using psql and Python.
+
+### Use psql in Terminal
+
+To securely interact with OpenAI through pgai from Terminal:
+
+1. Set your OpenAI key as an environment variable:
+    ```bash
+    OPENAI_AI_KEY="this-is-my-super-secret-api-key-dont-tell"
+    ```
+
+1. Set the value of your environment variable as a [psql variable](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-VARIABLES).
+   ```bash
+    psql -v OPENAI_API_KEY=$OPENAI_API_KEY
+    ```
+   You can now connect to your database using your psql variable as a [command line argument](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-OPTION-VARIABLE).
+
+1. Pass your API key as a parameterized variable when you query the database:
+    ```sql
+    SELECT * 
+    FROM openai_list_models($1)
+    ORDER BY created DESC
+    \bind :OPENAI_API_KEY
+    \g
+    ```
+
+    You use [bind](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-BIND) to pass the value of `OPENAI_API_KEY` as a parameterized variable. Now, the value 
+    of your API key is not visible in the query, logs, pg_stat_statements, and so on.
 
-### Creating the extension
+The `\bind` metacommand is available in psql version 16+.
+
+
+
+### Connect through psql in a Python app
+
+To securely interact with OpenAI in a Python app, you use [dotenv](https://github.com/theskumar/python-dotenv) to load API tokens, passwords, connection 
+strings, and other configuration settings from environment variables and .env files. You then pass your security 
+information as variables when you interact with your database.
+
+1. In your Python environment, include the dotenv packages:
+
+    ```bash
+    pip install python-dotenv
+    pip install psycopg2-binary
+    ```
+    TODO: could we not install these packages with the install scripts?
```

Comment:
These packages need to be installed on the client, not on the server. A user may choose to use Python or virtually any other programming language to connect to postgres and use the pgai extension. A python example was included because it is an especially popular language in the LLM realm.